src/test_utils.jl

```diff
@@ -66,60 +66,96 @@ function FakeTemplate{PointEstimate, MultiOutput}()
+and identity covariance matrix) for each observation.
+"""
+function FakeTemplate{DistributionEstimate, MultiOutput, DistributionInject}()
+    FakeTemplate{DistributionEstimate, MultiOutput, DistributionInject}() do num_variates, inputs
+        std_dev = ones(num_variates)
+        means = mean.(inputs)
+        return [Product(Normal.(m, std_dev)) for m in means]
+    end
+end
+
+"""
+    FakeTemplate{DistributionEstimate, MultiOutput, PointOrDistributionInject}()
+
+A [`Template`](@ref) whose [`Model`](@ref) will accept real-valued or distribution inputs to 
+predict a multivariate normal distribution (with means matching those of the passed 
+observations and identity covariance matrix) for each observation.
+"""
+function FakeTemplate{DistributionEstimate, MultiOutput, PointOrDistributionInject}()
+    FakeTemplate{DistributionEstimate, MultiOutput, PointOrDistributionInject}() do num_variates, inputs
+        std_dev = ones(num_variates)
+        means = _handle_inputs(inputs)
+        return [Product(Normal.(m, std_dev)) for m in means]
+    end
+end
+
+_handle_inputs(inputs::AbstractVector{<:Normal}) = mean.(inputs)
+_handle_inputs(inputs::AbstractMatrix) = [mean(inputs[m, :]) for m in 1:size(inputs, 2)]
+
 """
     FakeModel
 
 A fake Model for testing purposes. See [`FakeTemplate`](@ref) for details.
 """
-mutable struct FakeModel{E<:EstimateTrait, O<:OutputTrait} <: Model
+mutable struct FakeModel{E<:EstimateTrait, O<:OutputTrait, I<:InjectTrait} <: Model
     predictor::Function
     num_variates::Int
 end
 
-Models.estimate_type(::Type{<:FakeModel{E, O}}) where {E, O} = E
-Models.output_type(::Type{<:FakeModel{E, O}}) where {E, O} = O
+Models.estimate_type(::Type{<:FakeModel{E, O, I}}) where {E, O, I} = E
+Models.output_type(::Type{<:FakeModel{E, O, I}}) where {E, O, I} = O
+Models.inject_type(::Type{<:FakeModel{E, O, I}}) where {E, O, I} = I
 
-Models.estimate_type(::Type{<:FakeTemplate{E, O}}) where {E, O} = E
-Models.output_type(::Type{<:FakeTemplate{E, O}}) where {E, O} = O
+Models.estimate_type(::Type{<:FakeTemplate{E, O, I}}) where {E, O, I} = E
+Models.output_type(::Type{<:FakeTemplate{E, O, I}}) where {E, O, I} = O
+Models.inject_type(::Type{<:FakeTemplate{E, O, I}}) where {E, O, I} = I
 
 function StatsBase.fit(
-    template::FakeTemplate{E, O},
+    template::FakeTemplate{E, O, I},
     outputs,
     inputs,
     weights=uweights(Float32, size(outputs, 2))
-) where {E, O}
+) where {E, O, I}
     outputs = NamedDimsArray{(:variates, :observations)}(outputs)
     num_variates = size(outputs, :variates)
-    return FakeModel{E, O}(template.predictor, num_variates)
+    return FakeModel{E, O, I}(template.predictor, num_variates)
 end
 
-StatsBase.predict(m::FakeModel, inputs) = m.predictor(m.num_variates, inputs)
+StatsBase.predict(m::FakeModel, inputs::AbstractMatrix) = m.predictor(m.num_variates, inputs)
+StatsBase.predict(m::FakeModel, inputs::AbstractVector{<:Normal}) = m.predictor(m.num_variates, inputs)
```

Comment:
This `inputs::AbstractVector{<:Normal}` is more restrictive than it probably should be.  What is the most general requirement on the `Distribution` here?  Might `inputs::AbstractVector{<:Sampleable}` be appropriate?